test/framework/src/test/java/org/elasticsearch/test/disruption/NetworkDisruptionIT.java

```diff
@@ -102,4 +112,56 @@ public void testNetworkPartitionRemovalRestoresConnections() throws Exception {
         }
     }
 
+    public void testTransportRespondsEventually() throws InterruptedException {
+        internalCluster().setBootstrapMasterNodeIndex(0);
+        internalCluster().ensureAtLeastNumDataNodes(randomIntBetween(3, 5));
+        final NetworkDisruption.DisruptedLinks disruptedLinks;
+        if (randomBoolean()) {
+            disruptedLinks = TwoPartitions.random(random(), internalCluster().getNodeNames());
+        } else {
+            disruptedLinks = NetworkDisruption.Bridge.random(random(), internalCluster().getNodeNames());
+        }
+
+        NetworkDisruption networkDisruption = new NetworkDisruption(disruptedLinks, randomFrom(new NetworkDisruption.NetworkUnresponsive(),
+            new NetworkDisruption.NetworkDisconnect(), NetworkDisruption.NetworkDelay.random(random())));
+        internalCluster().setDisruptionScheme(networkDisruption);
+
+        networkDisruption.startDisrupting();
+
+        CountDownLatch latch = new CountDownLatch(100);
+        for (int i = 0; i < 100; ++i) {
+            TransportService transportService = internalCluster().getInstance(TransportService.class);
+            TransportService target = internalCluster().getInstance(TransportService.class);
+
+            transportService.sendRequest(target.getLocalNode(), ClusterHealthAction.NAME, new ClusterHealthRequest(),
+                new TransportResponseHandler<>() {
+                @Override
+                public void handleResponse(TransportResponse response) {
+                    latch.countDown();
+                }
+
+                @Override
+                public void handleException(TransportException exp) {
+                    latch.countDown();
+                }
+
+                @Override
+                public String executor() {
+                    return ThreadPool.Names.SAME;
+                }
+
+                @Override
+                public TransportResponse read(StreamInput in) throws IOException {
+                    return ClusterHealthResponse.readResponseFrom(in);
+                }
+            });
+        }
+
+        // give a bit of time to send something under disruption.
+        latch.await(500, TimeUnit.MILLISECONDS);
```

Comment:
Is there a way we could assert that this _does_ time out?

```suggestion
        assertFalse(latch.await(500, TimeUnit.MILLISECONDS));
```

However this won't work if none the messages we send are across disrupted links. Can we try and ensure that there is always at least one disrupted message?

Reply:
The `NetworkDisconnect` disruption type notifies of failures immediately. I would like to keep that disruption type included in the test. I ended up conditionally checking that we time out only when disruption type is not `NetworkDisconnect`, see: 779b98a2a1b7e88a22b50237db122ff59a8e1ff5.

It does kind of complicate the test somewhat and I am considering if it is better without it, WDYT?

Reply:
I see what you mean, but I do prefer it as it is now (779b98a) just in case we ever accidentally make this test vacuous.